js/private/js_library.bzl

```diff
@@ -143,23 +143,19 @@ target in {file_basename}'s package and add that target to the deps of {this_tar
         else:
             sources.append(file)
 
-    # sources
-    sources.extend([
-        item
+    # sources as depset
+    sources = depset(sources, transitive = [
+        target[JsInfo].sources
         for target in targets
-        if JsInfo in target and hasattr(target[JsInfo], "sources")
```

Comment:
Would `JsInfo.sources` being `None` and keeping this `if` be of any value? Or just annoying?

Reply:
Nice catch. It is worth keeping the hasattr as a defence in depth